src/ConfigEditor/ConfigEditor.tsx

```diff
@@ -113,10 +161,24 @@ export function ConfigEditor(props: Props) {
     return `${clusterEndpoint}/${jsonData.database || ''}`;
   };
 
+  const getWorkgroupUrl = async (workgroupName: string) => {
+    const { jsonData } = props.options;
+    if (workgroupName !== jsonData.workgroupName || workgroupEndpoint === '') {
+      const workgroups = await fetchWorkgroups();
+      const endpoint = workgroups.find((w) => w.value === workgroupName)?.description || workgroupName;
+      setWorkgroupEndpoint(endpoint || workgroupName);
+      return `${endpoint}/${jsonData.database || ''}`;
+    }
+    return `${workgroupEndpoint}/${jsonData.database || ''}`;
+  };
+
   useEffect(() => {
-    if (props.options.jsonData.clusterIdentifier) {
+    if (!props.options.jsonData.useServerless && props.options.jsonData.clusterIdentifier) {
       getClusterUrl(props.options.jsonData.clusterIdentifier);
     }
+    if (props.options.jsonData.useServerless && props.options.jsonData.workgroupName) {
+      getWorkgroupUrl(props.options.jsonData.workgroupName);
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
```

Comment:
would we want to recall this function if useServerless changes?

Reply:
IMO it's not necessarily because AWS resources like clusters, workgroups and secrets are less likely to change in a short period using ConfigEditor. 
Also current plugin does not recall fetchSecrets if useManagedSecret changes. 

If we want to recall more often, it can be a different PR.